### PR TITLE
Epic 7.2: Extended File Support for Flashcard Generator - Digital Guardians

### PR DESCRIPTION
**Pull request by dinglunz**
**Standardized Document Loading:**

- Custom loaders now yield Langchain Document objects as a list, ensuring compatibility with the load_and_summarize chain.

- Various sub-loaders are implemented for handling different document types such as PDFs, PowerPoint presentations, text files, JSON files, Markdown files, DOCX files, Azure documents, CSV files, HTML files, Google Sheets, and Google Slides.

**Text Splitting for Effective Summarization:**

- Text from documents is appropriately segmented using the RecursiveCharacterTextSplitter.

- The chunk size parameters are optimized for each type of document to ensure effective summarization.

**Integration with Map-Reduce Algorithm:**

- The load_summarize_chain from Langchain is utilized with the Map Reduce algorithm to handle the summarization of multiple documents.

- The chain processes both YouTube transcripts and documents from custom loaders, ensuring comprehensive summarization.

**Enhanced Summarization Process:**

- The summarization chain is enhanced to handle large document sets by selecting the K-nearest neighbors based on semantic relevance.

- Document embeddings are computed using OpenAIEmbeddings, and the most relevant documents are selected for summarization.

**Evaluation of Summary Quality:**

- The summary quality is evaluated by monitoring the number of batches created and the length of the summary.

- Edge cases, such as handling a large number of documents, are addressed by selecting the most relevant documents for summarization, preventing the Map Reduce algorithm from breaking.

### CODE REVIEW

File: app/features/dynamo/tools.py [hunk 1]
```diff
@@ -27,6 +55,439 @@ def read_text_file(file_path):
...
+        self.file_content = file_content
+        self.file_type = file_type
+
+    def load(self) -> List[Document]:
+        try:
+            content = self.file_content.read().decode('utf-8')
+            doc = Document(page_content=content, metadata={"source": self.file_type})
+            return [doc]
+        except Exception as e:
+            logger.error(f"Failed to load Markdown file from {self.file_type}")
```
Comment (mikhailocampo) [inline app/features/dynamo/tools.py:149]: Ensure the behavior of the try-except blocks are the same. In this case, it is returning an empty list while other classes' load method is raising a ValueError

**Review by mikhailocampo — COMMENTED** [review]
See changes requested

**Review by hash2004 — CHANGES_REQUESTED** [review]
Please review the changes requested by Mikhael.

**Comment by AaronSosaRamos** [comment]
Closed due inactivity

**Pull request closed without merging**